src/runtime/offers-flow.js

```diff
@@ -74,34 +109,47 @@ export class OffersFlow {
    * @return {!Promise}
    */
   start() {
-    // Start/cancel events.
-    this.deps_.callbacks().triggerFlowStarted(SubscriptionFlows.SHOW_OFFERS);
-    this.activityIframeView_.onCancel(() => {
-      this.deps_.callbacks().triggerFlowCanceled(SubscriptionFlows.SHOW_OFFERS);
-    });
+    if (this.activityIframeView_) {
+      // so no error is thrown if offers skipped
+      // Start/cancel events.
+      this.deps_.callbacks().triggerFlowStarted(SubscriptionFlows.SHOW_OFFERS);
+      this.activityIframeView_.onCancel(() => {
+        this.deps_
+          .callbacks()
+          .triggerFlowCanceled(SubscriptionFlows.SHOW_OFFERS);
+      });
 
-    // If result is due to OfferSelection, redirect to payments.
-    this.activityIframeView_.onMessageDeprecated(result => {
-      if (result['alreadySubscribed']) {
-        this.deps_.callbacks().triggerLoginRequest({
-          linkRequested: !!result['linkRequested'],
-        });
-        return;
-      }
-      if (result['sku']) {
-        new PayStartFlow(
-          this.deps_,
-          /** @type {string} */ (result['sku'])
-        ).start();
-        return;
-      }
-      if (result['native']) {
-        this.deps_.callbacks().triggerSubscribeRequest();
-        return;
-      }
-    });
+      // If result is due to OfferSelection, redirect to payments.
+      this.activityIframeView_.onMessageDeprecated(result => {
+        if (result['alreadySubscribed']) {
+          this.deps_.callbacks().triggerLoginRequest({
+            linkRequested: !!result['linkRequested'],
+          });
+          return new Promise();
```

Comment:
Why are you returning New Promises now?
Previously there were just return statements.

Reply:
I'm returning promises here because the compiler yelled at me if I didn't return promises here, since the place that invoked this function expected a promise to be returned.

Reply:
(happened to see this in email)

Did you mean return Promise.resolve()?  As in this promise is finished and complete now?

Reply:
Oh, I'd tried `return new Promise.resolve()` and it didn't work, but in retrospect I probably should have done it without new. I think what you're suggesting would be better than what I have now.

Reply:
I've reset them to just say `return;` for now, since that really makes the most sense. Will re-evaluate depending on whether it passes presubmit or not.